### ISSUE TITLE
skeletonize Loss of information

### ISSUE THREAD

**Issue by chl916185** [scikit-image/scikit-image#6321]
![image](https://user-images.githubusercontent.com/28615358/162139697-5f2db1ae-3d20-4acc-af02-d55c16c37b02.png)
I get a point？？？

**Comment by chl916185** [comment]
![skeleton_ori](https://user-images.githubusercontent.com/28615358/162140343-73f7809d-a199-4425-95aa-8d4f7e080263.png)
this is my image, you can try it.

**Comment by jni** [comment]
Yeah I can verify this behaviour on main. I don't know the reason yet though.

**Comment by chl916185** [comment]
> Yeah I can verify this behaviour on main. I don't know the reason yet though.

The latest version is 0.19.2 ，it has the same problem,Looking forward to your solution，thanks！

**Comment by rfezzani** [comment]
I can reproduce the observed behavior with current dev version:
```python
from skimage.morphology import skeletonize
import imageio
img = imageio.imread("/tmp/img.png")
sk = skeletonize(img > 0)
fig, ax = plt.subplots(1, 1, figsize=(8, 8))
ax.imshow((img > 0).astype(int) + sk.astype(int), "hot")
ax.set_axis_off()
fig.tight_layout()
```
![res](https://user-images.githubusercontent.com/3438227/162148322-6b3631d5-e732-4187-9858-2cc7f0c2a568.png)

In the resulting image, white pixels corresponds to `skeletonize` output and orange pixels are removed from the input image.

I don't know if it is an expected result, but the different behaviors between the two parallel lines is suspicious :/

BTW, the [documentation](https://scikit-image.org/docs/stable/api/skimage.morphology.html#skimage.morphology.skeletonize) of `skeletonize` is not correct:
> A binary image containing the objects to be skeletonized. Zeros represent background, nonzero values are foreground.

But I get a `ValueError: Image contains values other than 0 and 1` when I pass input image directly :(

**Comment by chl916185** [comment]
> I can reproduce the observed behavior with current dev version:
> 
> ```python
> from skimage.morphology import skeletonize
> import imageio
> img = imageio.imread("/tmp/img.png")
> sk = skeletonize(img > 0)
> fig, ax = plt.subplots(1, 1, figsize=(8, 8))
> ax.imshow((img > 0).astype(int) + sk.astype(int), "hot")
> ax.set_axis_off()
> fig.tight_layout()
> ```
> 
> ![res](https://user-images.githubusercontent.com/3438227/162148322-6b3631d5-e732-4187-9858-2cc7f0c2a568.png)
> 
> In the resulting image, white pixels corresponds to `skeletonize` output and orange pixels are removed from the input image.
> 
> I don't know if it is an expected result, but the different behaviors between the two parallel lines is suspicious :/
> 
> BTW, the [documentation](https://scikit-image.org/docs/stable/api/skimage.morphology.html#skimage.morphology.skeletonize) of `skeletonize` is not correct:
> 
> > A binary image containing the objects to be skeletonized. Zeros represent background, nonzero values are foreground.
> 
> But I get a `ValueError: Image contains values other than 0 and 1` when I pass input image directly :(

![image](https://user-images.githubusercontent.com/28615358/162151661-3b80f2ff-8f24-45d5-b67e-cd17eae6c3d5.png)
Why did you delete so many points? Annoying!

**Comment by rfezzani** [comment]
As a workaround, you can use Lee's method:
```python
sk = skeletonize(img, method="lee")
```
![res_lee](https://user-images.githubusercontent.com/3438227/162158592-52af5b58-a567-47ce-945f-a564042cb127.png)

**Comment by rfezzani** [comment]
@chl916185, the behavior you observed is an identified weakness of Zhang's method, please see [`A Comment on “A Fast Algorithm for Thinning Digital Patterns”`](https://www.csie.ntu.edu.tw/~hil/paper/cacm86.pdf)(third page, second point)
> Structures that should be preserved are destroyed. This can be seen in Figure 7. In general, diagonal segments with thickness 2 will eventually be reduced to only one or two points as shown in Figure 8.

**Comment by jni** [comment]
Interesting, thanks for digging @rfezzani! In that case, if the fault lies with the algorithm and not our implementation, and we should close the issue.

@chl916185, as @rfezzani writes above, your issue is the result of a fundamental limitation of the algorithm, and can be addressed by using a different algorithm. You can also perhaps [dilate](https://scikit-image.org/docs/stable/api/skimage.morphology.html#skimage.morphology.binary_dilation) your image before skeletonizing.
